File: W2D2-Activities2/app/src/main/java/com/example/activities/MainActivity.java
package com.example.activities;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText editTextUrl = findViewById(R.id.editTextUrl);
        Button buttonGetSecondActivity = findViewById(R.id.buttonGetSecondActivity);

        buttonGetSecondActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                String url = editTextUrl.getText().toString();
                Intent myIntent = new Intent(MainActivity.this,SecondActivity.class);
                myIntent.putExtra("urlKey",url);
               // myIntent.putExtra("testKey", "TEST");
                startActivity(myIntent);
            }
        });

        Log.d("Main Activity Life Cycle:", "onCreate");

    }

    @Override
    protected void  onStart(){
        super.onStart();
        Log.d("Main Activity Life Cycle:", "onStart");
    }



    @Override
    protected void  onResume(){
        super.onResume();
        Log.d("Main Activity Life Cycle:", "onResume");
    }


    @Override
    protected void  onPause(){
        super.onPause();
        Log.d("Main Activity Life Cycle:", "onPause");
    }


    @Override
    protected void  onStop(){
        super.onStop();
        Log.d("Main Activity Life Cycle:", "onStop");
    }


    @Override
    protected void  onDestroy(){
        super.onDestroy();
        Log.e("Main Activity Life Cycle:", "onDestroy");
    }
}

File: W10D5-MarsEstate/app/src/main/java/com/example/marsapp/view/MainActivity.kt
package com.example.marsapp.view

import android.content.Context
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.RecyclerView
import com.example.marsapp.R
import com.example.marsapp.data.MarsResponseItem

import com.example.marsapp.repo.MarsPropertyRepostory
import com.example.marsapp.view.adapters.MarsPropertyAdapter


class MainActivity : AppCompatActivity() {

    private val viewModel: MainViewModel by viewModels()
    private lateinit var recyclerView: RecyclerView


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        MarsPropertyRepostory.initialize(this)

        recyclerView = findViewById(R.id.recycler_view)

        val buttonBuy = findViewById<Button>(R.id.buttonBuy)
        val buttonRent = findViewById<Button>(R.id.buttonRent)
        val buttonPrice = findViewById<Button>(R.id.buttonPrice)
        val editTextPrice = findViewById<EditText>(R.id.editTextPrice)


        buttonBuy.setOnClickListener {
            viewModel.getFilterServiceCall("buy")
        }

        buttonRent.setOnClickListener {
            viewModel.getFilterServiceCall("rent")
        }

        buttonPrice.setOnClickListener {
            val price = editTextPrice.text.toString().toInt()
            viewModel.filterWithPrice(price)
        }

        //viewModel.marsServiceCall()

        viewModel.properties.observe(this, Observer { marsResponse ->
            setupRecyclerView(marsResponse)
            /*
                Log.e("MainActivityServistenGelen", marsResponse.toString())
                Glide.with(this)
                    .load(marsResponse[0].imgSrc) // image url
                    .centerCrop()
                    .into(imageView)
            */

        })
    }

    private fun setupRecyclerView(marsProperties: List<MarsResponseItem>) {
        val adapter = MarsPropertyAdapter(marsProperties)
        recyclerView.adapter = adapter
    }
}